apiVersion: openapi/v1alpha1
kind: Radar
metadata:
    name: "11950976"
    labels:
        datastore_id: "1845401"
data:
    classification: Serious Bug
    created: "2012-07-25T00:58:57.24438Z"
    description: "Summary:\r\n\r\nObjC++ code fails to catch exceptions by baseclass when run from a \"Foundation\" tool/app. Same code works when run from a pure \"C++\" tool/app.\r\n\r\nSteps to Reproduce:\r\n\r\n1. Embed this code snippet into a \"Foundation\" command-line tool project:\r\n\r\n\t#include <iostream>\r\n\t#include <stdexcept>\r\n\t\r\n\t//. . .\r\n\t\r\n\ttry {\r\n\t\tthrow std::invalid_argument(\"Ecky ecky ecky ecky pakang zoom boing!\");\r\n\t}\r\n\tcatch(const std::exception &e) {\r\n\t\tstd::cout << \"Exception raised: \" << e.what() << std::endl;\r\n\t}\r\n\tcatch( ... ){\r\n\t    std::cout << \"Unknown exception raised!\" << std::endl;\r\n\t}\r\n\t\r\n\ttry {\r\n\t\tthrow std::invalid_argument(\"Ecky ecky ecky ecky pakang zoom boing!\");\r\n\t}\r\n\tcatch(const std::invalid_argument &e) {\r\n\t\tstd::cout << \"Exception raised: \" << e.what() << std::endl;\r\n\t}\r\n\tcatch( ... ){\r\n\t    std::cout << \"Unknown exception raised!\" << std::endl;\r\n\t}\r\n\t\r\n2. Run it.\r\n\r\nExpected Results:\r\n\r\n\t// Correct output:\r\n\t// Exception raised: Ecky ecky ecky ecky pakang zoom boing!\r\n\t// Exception raised: Ecky ecky ecky ecky pakang zoom boing!\r\n\r\nActual Results:\r\n\r\n\t// Wrong output:\r\n\t// Unknown exception raised!\r\n\t// Exception raised: Ecky ecky ecky ecky pakang zoom boing!\r\n\r\nDoing the same within a \"C++\"  command-line tool project results in correct output. Same code. No changes.\r\n\r\nNotes:\r\nFound this stack overflow question dealing with the same issue:\r\nhttp://stackoverflow.com/questions/10602726/xcode-llvm-catch-clause-not-matching-derived-types"
    email: satsumac@googlemail.com
    modified: "2012-07-25T00:58:57.245352Z"
    number: "11950976"
    number_intvalue: 11950976
    originated: 25-Jul-2012
    parent_number: '&{NULL_VALUE}'
    product: Developer Tools
    product_version: 4.3.3+
    reproducible: Always
    resolved: ""
    status: ""
    title: ObjC++ code fails to catch exceptions by baseclass, works in pure C++
